src/styles/main.scss

```diff
@@ -1,3 +1,56 @@
+@import 'variable.scss';
+
 body {
-  margin: 0;
+  font-family: $numberFont;
+  font-size: $numberSize;
+}
+.calendar {
+  display: flex;
+  flex-wrap: wrap;
+  justify-content: left;
+  box-sizing: border-box;
+  margin: 40px auto;
+  max-width: ($cellSize * 7) + ($gap * 6);
+
+  @for $i from 0 to length($monthLengths) {
+    $length: nth($monthLengths, $i + 1);
+    &--month-length-#{$length} .day:nth-child(n + #{$length + 1}) {
+      display: none;
+    }
+  }
+
+  @each $day, $index in $daysOfWeek {
+    &--start--#{$day} {
+
+      .calendar__day:first-child {
+       margin-left: ($cellSize * $index);
+      }
+    }
+  }
+
+  &__day {
+    width: $cellSize;
+    height: $cellSize;
+    border: 1px solid $borderColor;
+    background-color: $backgroundColor;
+    box-sizing: border-box;
+    display: flex;
+    align-items: center;
+    justify-content: center;
+    position: relative;
+
+    @for $index from 1 to 32 {
+      &:nth-child(#{$index})::before {
+        content: "#{$index}";
+
+      }
+    }
+
+    &:hover {
+      cursor: pointer;
+      background-color: pink;
+      transform: translateY(-20px);
+      transition: transform $transitionDuration;
```

Comment:
Make both 'hover' and 'unhover' be smooth. 
To reach this effect you should apply `transition` not to `:hover` but to static selector
Wrong Example: 
```
.button {
  color: #ff000;
  
  &:hover {
    color: #00ff00;
    transition: color 250ms;
  }
}
```

Correct Example:
```
.button {
  color: #ff000;
  transition: color 250ms;
      
  &:hover {
    color: #00ff00;
  }
}
```

Reply:
not fixed